library/gen_x12_837.inc.php

```diff
@@ -1043,95 +1043,98 @@ function gen_x12_837($pid, $encounter, &$log, $encounter_claim = false)
-        // Segment REF (Purchased Service Provider Secondary Identification) omitted.
-        // Segment NM1,N3,N4 (Service Facility Location) omitted.
-        // Segment REF (Service Facility Location Secondary Identification) omitted.
-        // Segment NM1 (Supervising Provider Name) omitted.
-        // Segment REF (Supervising Provider Secondary Identification) omitted.
-        // Segment NM1,N3,N4 (Ordering Provider) omitted.
-        // Segment REF (Ordering Provider Secondary Identification) omitted.
-        // Segment PER (Ordering Provider Contact Information) omitted.
-        // Segment NM1 (Referring Provider Name) omitted.
-        // Segment REF (Referring Provider Secondary Identification) omitted.
-        // Segments NM1*PW, N3, N4 (Ambulance Pick-Up Location) omitted.
-        // Segments NM1*45, N3, N4 (Ambulance Drop-Off Location) omitted.
-
-        // REF*1C is required here for the Medicare provider number if NPI was
-        // specified in NM109.  Not sure if other payers require anything here.
-        ++$edicount;
-        $out .= "REF" .
-            "*" . $claim->providerNumberType($prockey) .
-            // Note: 5010 documents that type 1D (Medicaid) is changed to G2.
-            "*" . $claim->providerNumber($prockey) .
-            "~\n";
+            if ($claim->providerTaxonomy($prockey)) {
+                ++$edicount;
+                $out .= "PRV" .
+                "*" . "PE" . // PErforming provider
+                "*" . "PXC" .
+                "*" . $claim->providerTaxonomy($prockey) .
+                "~\n";
+            }
+
+            // Segment PRV*PE (Rendering Provider Specialty Information) omitted.
+            // Segment REF (Rendering Provider Secondary Identification) omitted.
+            // Segment NM1 (Purchased Service Provider Name) omitted.
+            // Segment REF (Purchased Service Provider Secondary Identification) omitted.
+            // Segment NM1,N3,N4 (Service Facility Location) omitted.
+            // Segment REF (Service Facility Location Secondary Identification) omitted.
+            // Segment NM1 (Supervising Provider Name) omitted.
+            // Segment REF (Supervising Provider Secondary Identification) omitted.
+            // Segment NM1,N3,N4 (Ordering Provider) omitted.
+            // Segment REF (Ordering Provider Secondary Identification) omitted.
+            // Segment PER (Ordering Provider Contact Information) omitted.
+            // Segment NM1 (Referring Provider Name) omitted.
+            // Segment REF (Referring Provider Secondary Identification) omitted.
+            // Segments NM1*PW, N3, N4 (Ambulance Pick-Up Location) omitted.
+            // Segments NM1*45, N3, N4 (Ambulance Drop-Off Location) omitted.
+
+            // REF*1C is required here for the Medicare provider number if NPI was
+            // specified in NM109.  Not sure if other payers require anything here.
+
+            // @TODO below appears to not belong in 5010??.
```

Comment:
the 3 comments and blank line in @bradymiller post can be removed
the comments above that are helpful